tcf_website/management/commands/upload_csv_review_drive.py

```diff
@@ -0,0 +1,213 @@
+
+from django.core.management.base import BaseCommand
+
+from tcf_website.models import *
+
+DATA_DIR = "tcf_website/management/commands/review_drive_responses/"
+
+
+class Command(BaseCommand):
+
+    # Run this coomand using `sudo docker exec -it tcf_django python3 manage.py upload_csv_review_drive [filename] [season] [year]`
+    # Optional flag --verbose can be set to show output of the script
+    # Must be run in a separate terminal after already running docker-compose up
+    # May have to restart the docker container for database changes to be visible on the site
+    # Visually check that CS department has been moved to E School and that
+    # other schools have been split up.
+
+    help = (
+        "Uploads CSV from /review_drive_responses/ in the format of Time, Email, Mnemonic, Number, Name, Review, "
+        "Rating, Enjoyable, Recommendation, Difficulty, Reading Hours, Writing Hours, Group work Hours, Misc Hours"
+    )
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "filename",
+            help=(
+                "Filepath of csv to upload in /review_drive_responses/"
+                "Must be in correct format."
+            ),
+            type=str,
+        )
+        parser.add_argument(
+            "season",
+            help=(
+                "What season the reviews took place i.e. Fall, January, Spring, Summer"
+            ),
+            type=str,
+        )
+        parser.add_argument(
+            "year",
+            help=("Year of reviews i.e. 2024"),
+            type=int,
+        )
+
+        # Named (optional) arguments
+        parser.add_argument(
+            "--verbose",
+            action="store_true",
+            help="Verbose output",
+        )
+
+    def handle(self, *args, **options):
+        verbose = options["verbose"]
+        filename = options["filename"]
+        season = options["season"]
+        year = options["year"]
+
+        # Get semester object from args
+        semester = Semester.objects.filter(
+            season=season.upper(), year=year
+        ).first()
+        if semester.count() <= 0:
+            print("ERROR: Semester not found")
+            return
+
+        # Needs to be created ahead of time in the db
+        dummy_account = User.objects.filter(computing_id__iexact="reviewdrive")
+        if dummy_account.count() <= 0:
+            print("ERROR: DUMMY ACCOUNT NOT FOUND")
+            return
+
+        if verbose:
+            print("Starting file upload...")
+
+        file = open(os.path.join(DATA_DIR, filename), mode="r")
+        csv_file = csv.reader(file)
+
+        for i, line in enumerate(csv_file):
+            if i == 0:
+                continue
+
+            # Strip needed to remove random whitespace for exact match
+            email = line[1].strip()
+            mnemonic = line[2].strip()
+            num = int(line[3].strip())
+            instructor_arr = line[4].strip().split(" ", 1)
```

Comment:
this has been renamed from `instructor` but is not renamed elsewhere

Reply:
It is renamed, if it wasn't it would be called through instructor[0] or instructor[1]. instructor as a variable is now the object called from the query.

Reply:
Seems like the `if verbose` print statement prints it before it is defined.

Reply:
@ajnye